kolibri/plugins/slideshow_render/assets/src/views/SlideshowRendererComponent.vue

```diff
@@ -0,0 +1,214 @@
+      <mat-svg v-else name="fullscreen" category="navigation" />
+    </UiIconButton>
+    <Hooper v-if="slides" @slide="handleSlide">
+      <Slide v-for="slide in slides" :key="slide.id">
+        <div
+          class="slideshow-slide-image"
+          :style="{
+            height: `calc(100% - ${caption_height}px)`
+          }"
+        >
+          <img
+            :src="slide.storage_url"
+            :aria-labelledby="slideTextId(slide.id)"
+          >
+        </div>
+        <div :id="slideTextId(slide.id)" class="hidden-descriptive-text">
+          {{ slide.descriptive_text || slide.caption }}
+        </div>
+        <div :ref="slide.id" class="caption">
+          {{ slide.caption }}
+        </div>
+      </Slide>
+      <HooperNavigation
+        slot="hooper-addons"
+        :class="{'hooper-navigation-fullscreen' : isInFullscreen}"
+      />
+      <HooperPagination slot="hooper-addons" />
+    </Hooper>
+  </CoreFullscreen>
+
+</template>
+
+
+<script>
+
+  import _ from 'lodash';
+
+  import themeMixin from 'kolibri.coreVue.mixins.themeMixin';
+  import contentRendererMixin from 'kolibri.coreVue.mixins.contentRendererMixin';
+  import responsiveElement from 'kolibri.coreVue.mixins.responsiveElement';
+  import responsiveWindow from 'kolibri.coreVue.mixins.responsiveWindow';
+
+  import UiIconButton from 'kolibri.coreVue.components.UiIconButton';
+  import CoreFullscreen from 'kolibri.coreVue.components.CoreFullscreen';
+  import {
+    Hooper,
+    Slide,
+    Navigation as HooperNavigation,
+    Pagination as HooperPagination,
+  } from 'hooper';
+  import '../../styles/custom-hooper.css';
+
+  export default {
+    name: 'SlideshowRendererComponent',
+    components: {
+      UiIconButton,
+      CoreFullscreen,
+      Hooper,
+      Slide,
+      HooperPagination,
+      HooperNavigation,
+    },
+    mixins: [contentRendererMixin, themeMixin, responsiveElement, responsiveWindow],
+    props: {
+      defaultFile: {
+        type: Object,
+        default: () => {},
+      },
+      files: {
+        type: Array,
+        default: () => [],
+      },
+    },
+    data: () => ({
+      isInFullscreen: false,
+      manifest: {},
+      slides: null,
+      currentSlide: null,
+      hooperSettings: {
+        centerMode: true,
+      },
+    }),
+    computed: {
+      slideshow_images: function() {
+        const files = this.files;
+        return files.filter(file => file.preset != 'Slideshow Manifest');
+      },
+      caption_height: function() {
+        return 30 + (this.currentSlide ? this.$refs[this.currentSlide.id][0].clientHeight : 0);
+      },
+    },
+    mounted() {
+      /*
+        Using the manifest file, get the JSON from the manifest, then
+        use the manifest JSON to get all slide images and metadata.
+      */
+      fetch(this.defaultFile.storage_url, { method: 'GET' })
+        .then(response => {
+          return response.status === 200 ? response.json() : {};
+        })
```

Comment:
@indirectlylit 

I refactored it to just return `response.json()` and added a `catch` block which dispatches `handleApiError` with the error object.

As for the `CORE_SET_PAGE_LOADING` - I tried:

```
mounted() {
      /*
        Using the manifest file, get the JSON from the manifest, then
        use the manifest JSON to get all slide images and metadata.
      */
      this.$store.commit("CORE_SET_PAGE_LOADING", true);
      fetch(this.defaultFile.storage_url, { method: 'GET' })
        .then(response => {
          return response.json();
        })
        .then(manifest_data => {
          this.manifest = manifest_data.slideshow_data;

          this.slides = orderBy(
            this.manifest.map(image => {
              return {
                storage_url: this.slideshowImages.find(
                  sFile => checksumFromFile(sFile) == image.checksum
                ).storage_url,
                caption: image.caption,
                sort_order: image.sort_order,
                id: image.checksum,
                descriptive_text: image.descriptive_text,
              };
            }),
            ['sort_order'],
            ['asc']
          );
          this.$store.commit("CORE_SET_PAGE_LOADING", true);
        })
        .catch(error => {
          this.$store.commit("CORE_SET_PAGE_LOADING", true);
          this.$store.dispatch('handleApiError', error);
        });
```

For some reason, this infinitely loops calling the `mounted()` method over and over. I looked a bit into that mutation and I'm not seeing how or why that would re-mount the component. 

Any thoughts on why this might be the case?

Also - FWIW - the functionality of the slideshow is to show a spinning loader until the images are all loaded. I've tested with bandwidth throttling to slow down the networking and it showed the spinner until it was fully loaded and ready.

Reply:
Sorry, should have caught this sooner - you should use the kolibri `client` module: `import client from 'kolibri.client';` to do any fetches (rather than using the `fetch` API which is not as widely supported: https://caniuse.com/#feat=fetch (e.g. IE11, but also many others).

Reply:
@rtibbles thanks for letting me know! Done and pushed.